src/routers/user.router.ts

```diff
@@ -50,6 +51,22 @@ async function initSession(user: UserEntity): Promise<{ did: string, appToken: s
 	};
 }
 
+async function filterUserData(user: UserEntity): Promise<{ id: number, did: string, appToken: string, username?: string, displayName: string, privateData: string, webauthnUserHandle: string, webauthnCredentials: WebauthnCredentialEntity[] }> {
+	const secret = new TextEncoder().encode(config.appSecret);
+	const appToken = await new SignJWT({ did: user.did })
+		.setProtectedHeader({ alg: "HS256" })
+		.sign(secret);
+	return {
+		id: user.id,
+		appToken,
+		did: user.did,
+		displayName: user.displayName || user.username,
+		privateData: user.privateData.toString(),
+		username: user.username,
+		webauthnUserHandle: user.webauthnUserHandle,
+		webauthnCredentials: user.webauthnCredentials,
+	};
+}
```

Comment:
Most of the data is duplicated between `filterUserData` and `initSession`... in fact it looks like `filterUserData` returns a strict superset of what `initSession` does (except they get distinct `appToken`s). Why do we need both? Especially since it looks like `filterUserData` is only ever used right next to `initSession`?

Reply:
I see two possible options:

1. Refactor the code to reduce duplicate parts.
2. Use only the filterUserData and separate the initSession logic on the frontend.

Which do you think is better? Or do you have another solution in mind?

Reply:
PR #59 is a good solution I think. :slightly_smiling_face: